manifests/init.pp

```diff
@@ -455,8 +455,7 @@
     notify  => Service['snmpd'],
   }
 
-  # Debian 9 use systemd
-  if ( $facts['os']['name'] == 'Debian' and versioncmp($facts['os']['release']['major'], '9') >= 0 ) {
+  if ( has_key($facts, 'service_provider') and $facts['service_provider'] == 'systemd' ) {
```

Comment:
we can even simplify this! there is the `fact()` method that will return undef if the requested fact does not exist. and undef is false in puppet DSL:
```suggestion
  if facts('service_provider') == 'systemd' {
```

Reply:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#fact